### PR TITLE
Clean up mcs build system

### PR DESCRIPTION
**Pull request by alexrp**
Please don't merge this before there's a green Jenkins build.

### CODE REVIEW

**Comment by akoeplinger** [comment]
:heart:

**Comment by alexrp** [comment]
build

File: mcs/tests/test-349.cs [hunk 1]
```diff
@@ -54,17 +54,7 @@ public static int Main ()
 		if ((typeof (X2).Attributes & TypeAttributes.BeforeFieldInit) == 0)
 			return 2;
 		
-#if NET_2_0
-		ConstructorInfo mi = typeof(C).GetConstructors ()[0];
```
Comment (marek-safar) [inline mcs/tests/test-349.cs]: This should not be removed

Comment (alexrp) [inline mcs/tests/test-349.cs]: The file doesn't specify `-d:NET_2_0` in its compiler options.

**Comment by migueldeicaza** [comment]
Alex, before we even merge this, we need a description of what is this attempting to solve.

I took a quick look and have a good idea, but I want to get a detailed explanation.

**Comment by alexrp** [comment]
@migueldeicaza I'll expand a bit on the commit messages, I just wanted to make sure this didn't break stuff left and right first.

**Comment by alexrp** [comment]
OK, I expanded on the commit messages, and also added a missing commit to remove some ancient 1.x assemblies.

**Comment by alexrp** [comment]
build

**Comment by alexrp** [comment]
@migueldeicaza this should be good to go. Let me know if there are any questions the commit messages don't address.

**Comment by migueldeicaza** [comment]
This looks good.

**Pull request merged**